### ISSUE TITLE
Export inside if statement with `plugin-replace`'d condition is undefined

### ISSUE THREAD

**Issue by Alduino** [PepsRyuu/nollup#192]
Minimal reproduction: https://github.com/Alduino/nollup-test-named-exports

Importing a named CommonJS export inside an if statement whose condition is a value replaced using `@rollup/plugin-replace` causes the value to be `undefined`.

See this example - in the Rollup config, the replace plugin is used to replace `REPLACED_VALUE` with `true`.

```js
// test-lib.js
if (REPLACED_VALUE) {
    exports.test = "test value";
}

// index.js
import {test} from "./test-lib";
console.log(test); // "test value" with Rollup, undefined with Nollup
```

This issue breaks React DOM, and that was the original problem for me - its main exports in `react-dom.development.js` are wrapped in an `if (process.env.NODE_ENV !== "production")`.

It also breaks `@emotion/cache` (i.e. `@emotion/react`) when they import `middleware` from `stylis`. Stylis's node module is minified so I'm not certain it's the same issue, but it is having the same effect so it's likely.

**Comment by PepsRyuu** [comment]
Thanks for the reproduction. Looking into it now.

**Comment by PepsRyuu** [comment]
Found the problem. It's an internal issue with Nollup transform sequence, where it loses ```syntheticNamedExports``` information if a module goes through an additional transformation hook. So what was happening is that ```commonjs``` went first, appended ```syntheticNamedExports``` to the module info, and then ```replace()``` executed, but it doesn't append that info so it was lost.

As a temporary workaround, swap ```commonjs``` and ```replace``` in the config so that ```replace``` comes first. I'll be working on a patch in the meantime which I should hopefully release either today or tomorrow.

**Comment by PepsRyuu** [comment]
Turned out to be easier than I thought. Released in ```0.16.1```. Enjoy!
